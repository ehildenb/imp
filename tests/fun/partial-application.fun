// 12

let f x = fun [ h : t ] h -> x + h
          |   l         z -> x * z
and apply f = f 4
 in apply (f 3 [ 5 : 6 : [ ] ])
 //in f 3 [ 5 : 6 : [ ] ] 4

// apply (f 3 [ 5 : 6 : [ ] ])
// f 3 [ 5 : 6 : [ ] ] 4
// 12

// Note if non-linear patterns aren't taken care of with the partial application properly, you get
// 7
