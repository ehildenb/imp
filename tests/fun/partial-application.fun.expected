<FUN>
  <k>
    12
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>
